### DISCUSSION TITLE
Reference id change

### DISCUSSION THREAD

**Discussion by paulgear** [pendulum-project/ntpd-rs#1076] [General]
Hi @davidv1992, I just noticed [Fixed incorrect id being used for our reference id](https://github.com/pendulum-project/ntpd-rs/commit/75c182eabb1df9be8caa8f08c33d64bcb5aaffa7) - is this to fix the situation where ntpd-rs returns the reference id of its upstream peer rather than that peer's IPv4 address or IPv6 hash?  I just noticed this behaviour and was going to submit an issue, but it seems like it might be fixed already.

**Comment by davidv1992** [comment]
Yes, it is a fix for that behaviour.

**Comment by paulgear** [comment]
Thanks!
